src/QMCWaveFunctions/tests/test_RotatedSPOsT.cpp

```diff
@@ -0,0 +1,911 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source
+// License. See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 QMCPACK developers.
+//
+// File developed by: Joshua Townsend, jptowns@sandia.gov, Sandia National
+// Laboratories
+//
+// File created by: Joshua Townsend, jptowns@sandia.gov, Sandia National
+// Laboratories
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "FakeSPOT.h"
+#include "OhmmsData/Libxml2Doc.h"
+#include "OhmmsPETE/OhmmsMatrix.h"
+#include "Particle/ParticleSetPoolT.h"
+#include "Particle/ParticleSetT.h"
+#include "QMCWaveFunctions/EinsplineSetBuilderT.h"
+#include "QMCWaveFunctions/RotatedSPOsT.h"
+#include "QMCWaveFunctions/WaveFunctionComponent.h"
+#include "catch.hpp"
+#include "checkMatrix.hpp"
+#include "type_traits/ConvertToReal.h"
+#include "type_traits/template_types.hpp"
+#include <ResourceCollection.h>
+#include <stdio.h>
+
+#include <limits>
+#include <string>
+#include <tuple>
+
+using std::string;
+
+namespace qmcplusplus
+{
+template<typename T>
+struct ValueApproxHelper
+{
+  using Type = Catch::Detail::Approx;
+};
+template<typename T>
+struct ValueApproxHelper<std::complex<T>>
+{
+  using Type = Catch::Detail::ComplexApprox;
+};
+
+template<typename T>
+using ValueApprox = typename ValueApproxHelper<T>::Type;
```

Comment:
This code might be useful elsewhere, and could get moved to src/external_codes/complex_approx.hpp or src/Configuration.h
(Not necessary for this PR, though)